slashing/program/src/state.rs

```diff
@@ -58,10 +58,15 @@ impl From<u8> for ProofType {
 
 /// Trait that proof accounts must satisfy in order to verify via the slashing
 /// program
-pub trait SlashingProofData {
+pub trait SlashingProofData<'a> {
```

Comment:
I've gotten burned by lifetimes in traits in the past, since they can make things complicated or totally unworkable without a lot of Rust magic. For example, I don't think you can have a type implement `SlashingProofData` where it copies the underlying data to own it.

I think we'll be ok in this case since it'll just be for unpacking things in a program, but something to be careful about.

Reply:
Guess it forces us to zerocopy everything in the future?